### DISCUSSION TITLE
Activate Templates on the Configure Template Page

### DISCUSSION THREAD

**Discussion by pmknowles** [arendst/Tasmota#15251] [Ideas]
Would it be possible to put an Activate Template checkbox on the Configure Template page?
A lot of people come onto the Google pages who have either entered a Template and not Activated it and can't understand why it isn't working or Activated a Template without realising and not understanding why Configuration changes won't take.
It doesn't make much sense for it to be on Other.
The same applies to the MQTT Enable checkbox (which also catches people out)

**Comment by sfromis** [comment]
Arguably, you should **rarely** use the page "Configure Template", unless you already have a template configured/activated. If people have an existing template to use, they should use "Configure Other".

**Comment by pmknowles** [comment]
Honestly?
You think that makes sense?
Oooh, I've got a Template.
Now where should I put it?
Ooh, I know, I'll ignore Configure Template and choose Configure Other...
And you ignored the point about MQTT Enable also confusing people.
Templates are designed to make it easy for people who don't know what they're doing. But then cause problems for them because they don't know what they're doing.
Personally, I think there should be an extra button in the Reset Configuration section which is Upload Template. You then enter the Template string (with an Apply which then populates the GPIO etc below it) or enter the GPIO data manually. Once you are happy with the settings, you then click Upload Template.
Further changes would then be made in the Configure Module pages.
I don't know how much time I have wasted trying to fix problems caused by people Templates - a Template should be used like in Word, once to set a device up, from then on you change the results in the device.
In fact, there's an argument for going further, with the Template having WiFi and MQTT fields so it becomes a configuration tool on the docs pages that you complete, download and then you could upload at the AP stage. There's more 'space' on the internet and more information could be given about the selections and what's allowed and not allowed (warnings about bad GPIO selections etc).

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
